monai/apps/reconstruction/mri_transforms.py

```diff
@@ -0,0 +1,301 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Sequence
+
+import numpy as np
+import torch
+from numpy import ndarray
+from torch import Tensor
+
+from monai.config.type_definitions import NdarrayOrTensor
+from monai.utils.type_conversion import convert_data_type, convert_to_tensor
+
+
+def convert_to_tensor_complex(data: ndarray) -> Tensor:
+    """
+    Convert numpy array to PyTorch tensor.
+    For complex arrays, the real and imaginary
+    parts are stacked along the last dimension.
+
+    Args:
+        data: Input numpy array
+
+    Returns:
+        PyTorch version of the data
+
+    Example:
+        .. code-block:: python
+
+            import numpy as np
+            data = np.array([ [1+1j, 1-1j], [2+2j, 2-2j] ])
+            # the following line prints (2,2)
+            print(data.shape)
+            # the following line prints torch.Size([2, 2, 2])
+            print(convert_to_tensor_complex(data).shape)
+    """
+    if np.iscomplexobj(data):
+        data = np.stack((data.real, data.imag), axis=-1)
+    return convert_data_type(data, torch.Tensor)[0]
+
+
+def complex_abs(x: NdarrayOrTensor) -> NdarrayOrTensor:
+    """
+    Compute the absolute value of a complex array.
+
+    Args:
+        x: Input array/tensor with 2 channels in the last dimension representing real and imaginary parts.
+
+    Returns:
+        Absolute value along the last dimention
+
+    Example:
+        .. code-block:: python
+
+            import numpy as np
+            x = np.array([3,4])[np.newaxis]
+            # the following line prints 5
+            print(complex_abs(x))
+    """
+    assert x.shape[-1] == 2
+    return (x[..., 0] ** 2 + x[..., 1] ** 2) ** 0.5
+
+
+class MaskFunc:
+    """
+    A basic class for under-sampling mask setup. It provides common features for under-sampling msak generators.
+    For example, RandomMaskFunc and EquispacedMaskFunc (two mask-generating objects defined right after this module)
+    both inherit MaskFunc to properly setup properties like the acceleration factor.
+    """
+
+    def __init__(self, center_fractions: Sequence[float], accelerations: Sequence[float]):
+        """
+        Args:
+            center_fractions: Fraction of low-frequency columns to be retained.
+                If multiple values are provided, then one of these numbers is chosen uniformly
```

Comment:
"If multiple" implies that single values can also be provided, which would mean that the typing should be `Union[float, Sequence[float]]`. Both should also be tested. You could use `ensure_tuple` such that if a float is provided, it is converted to a tuple of floats containing just a single element.

Reply:
hmm, the input here is expected to be a sequence. e.g., (2.0,) or [1.0, 3.0] are valid inputs.

Reply:
Ok, so always a sequence but could be a sequence containing just one value. That's fine!